Misc fixes

rpavlik opened the pull request:
These are miscellaneous fixes found in the process of setting up the CMake-based build system - though none of these touch the build system at all.

Some of them are warning fixes that are probably harmless, some are warning fixes that indicated a definite error, and some (like the plugin dialog) fix broken functionality. (I assumed that it was the way I was building that kept the plugin dialog from "expanding" the filter plugins, but it turns out it was actually a code defect, now fixed. See screenshot - filter_measure had no text below and no expander in the tree before these patches.)


![Screenshot from 2019-12-03 17-22-34](https://user-images.githubusercontent.com/61129/70098449-90458c00-15f1-11ea-8a76-36d0f1569dd4.png)
 

(Some of these warning fixes, etc. probably duplicate fixes in other PR's that haven't yet been merged.)

cc @jmespadero

jmespadero commented on src/meshlabplugins/filter_mutualglobal/alignGlobal.h:
```diff
@@ -30,7 +30,7 @@ Revision 1.1  2006/09/25 09:24:39  e_cerisoli
 
 ****************************************************************************/
-
+#pragma once
```
Beware that `#pragma once` , although supported by almost every modern compiler, is not C++ standard.

rpavlik replied:
Yes, I do know that unfortunately. (What compiler doesn't support it?) I can change this to a normal include if that better fits project rules.

jmespadero commented:
Thanks for all those changes....
 I have found two similar bugs in file `meshlabplugins/edit_align/meshtree.cpp`

```diff
diff --git a/src/meshlabplugins/edit_align/meshtree.cpp b/src/meshlabplugins/edit_align/meshtree.cpp
index 378e42e9..f14866ec 100644
--- a/src/meshlabplugins/edit_align/meshtree.cpp
+++ b/src/meshlabplugins/edit_align/meshtree.cpp
@@ -111,7 +111,7 @@ void MeshTree::ProcessArc(int fixId, int movId, vcg::Matrix44d &MovM, vcg::Align
 void MeshTree::Process(vcg::AlignPair::Param &ap, MeshTree::Param &mtp)
 {
   QString buf;
-  cb(0,qUtf8Printable(buf.sprintf("Starting Processing of %i glued meshes out of %i meshes\n",gluedNum(),nodeMap.size())));
+  cb(0,qUtf8Printable(buf.sprintf("Starting Processing of %i glued meshes out of %ld meshes\n",gluedNum(),nodeMap.size())));
 
   /******* Occupancy Grid Computation *************/
   cb(0,qUtf8Printable(buf.sprintf("Computing Overlaps %i glued meshes...\n",gluedNum() )));
@@ -179,7 +179,7 @@ void MeshTree::Process(vcg::AlignPair::Param &ap, MeshTree::Param &mtp)
 #pragma omp parallel for schedule(dynamic, 1) num_threads(num_max_thread)
   for(int i=0;i<totalArcNum; ++i)
   {
-    fprintf(stdout,"%4i -> %4i Area:%5i NormArea:%5.3f  %d\n",OG.SVA[i].s,OG.SVA[i].t,OG.SVA[i].area,OG.SVA[i].norm_area);
+    fprintf(stdout,"%4i -> %4i Area:%5i NormArea:%5.3f \n",OG.SVA[i].s,OG.SVA[i].t,OG.SVA[i].area,OG.SVA[i].norm_area);
     AlignPair::Result *curResult = findResult(OG.SVA[i].s,OG.SVA[i].t);
     if(curResult->err >= percentileThr) // missing arc and arc with great error must be recomputed.
     {
```

alemuntoni commented:
Thank you  very much @rpavlik for this huge pull request!
They all make sense to me, except the ```#pragma once``` @jmespadero mentioned in  ```filter_mutualglobal/alignGlobal.h```. It should be replaced with proper ```#ifdef``` guards.
If you can make this small change I'll then merge!

rpavlik commented:
I have adjusted the include guard, rebased, and also rebased on top of #387 (which similarly contained some fixes I had independently made at a later date), so merging this will also merge #387 effectively. (The original commit author credit there remains intact.)

jmespadero commented:
Give this man a medal.

alemuntoni commented:
This seems all ok now, thank you @rpavlik!!
